executor.go

```diff
@@ -230,7 +229,7 @@ func (e *Executor) Wait(ctx context.Context) (*ExecResult, error) {
 
 	res.Runtime = e.TimeRunning()
 
-	return res, exitErr
+	return res, err
 }
 
 // Run calls Start(), then Wait(), and returns an ExecResult and error (if
```

Comment:
I think we shall update this comment to explain a bit more about how shall a caller handle a non nil return like the exec result might still contains the actual commands exit status and outputs.